sqlglot/parser.py

```diff
@@ -5475,6 +5476,53 @@ def _parse_comprehension(
             condition=condition,
         )
 
+    def _parse_heredoc(self) -> t.Optional[exp.Heredoc]:
+        tags = []
+
+        if self._match_text_seq("$"):
+            tags.append("$")
+        else:
+            return None
+
+        if self._curr and self._is_connected():
+            if not self._match_text_seq("$"):
+                tag_text = self._curr.text
+                tags.append(tag_text)
+                self._advance()
+            else:
+                tag_text = None
+                tags.append("$")
+        else:
+            self.raise_error("No closing $ found")
+
+        if tag_text:
```

Comment:
tag_text can be undefined here because the raise_error on line 5496 doesn't necessarily break